### PR TITLE
P2fix

### PR DESCRIPTION
**Pull request by soulslicer**

### CODE REVIEW

File: python/sim.py [hunk 1]
```diff
@@ -181,7 +186,7 @@ def get_return(self, np_depth_image, np_design_pts):
         assert len(pylc_output.images_multi) == 1
         assert len(pylc_output.images_multi[0]) == 1
         # np.ndarray, dtype=np.float32, shape=(512, 512, 4)
-        output_image = pylc_output.images_multi[0][0]
+        output_image = pylc_output.images_multi[0][0].copy()
```
Comment (siddancha) [inline python/sim.py:189]: Is it good practice to copy objects returned from PyBind or something?

File: python/sim.py [hunk 2]
```diff
@@ -111,6 +113,9 @@ def __init__(self, LASER_PARAMS=None, CAMERA_PARAMS=None):
         c_datum.imgh = self.CAMERA_PARAMS['height']
         c_datum.imgw = self.CAMERA_PARAMS['width']
 
+        c_datum.hit_mode = self.CAMERA_PARAMS['hit_mode']
+        c_datum.hit_noise = self.CAMERA_PARAMS['hit_noise']
+
```
Comment (siddancha) [inline python/sim.py:118]: What is `hit_mode` and `hit_noise`?

**Review by siddancha — CHANGES_REQUESTED** [review]

File: python/sim.py (hunk 1 above)
Comment (soulslicer) [reply python/sim.py:189]: In this case, I suppose its not needed. output_image is pointing to pylc_output which is held in scope after the function ends as we are returning a part of it. eventually, output_image would go out of scope too, at which point the object should be deleted.

But with my experience with openpose, creating a copy before passing it to the user prevented certain weird scenarios from happening. especially if the user takes  the object and stores it, or passes it into his own function for manipulation etc.

File: python/sim.py (hunk 2 above)
Comment (soulslicer) [reply python/sim.py:118]: `hit_mode` is specifying the type of algorithm used to calculate the intensity. Right now there are two modes `0` and `1`, where the default is a linear function of the error to the distance on the depth map, and `1` is a non-linear/gaussian function. More algorithms may be possible in the future, such as taking into account normal or rgb information too

`hit_noise` is a a number from `0 to 1` that specifies the intensity standard deviation. Noise is sampled from a normal distribution with mean 0 and std `hit_noise` and added to the image intensity (where intensity is a value from 0 to 1 here)

File: python/sim.py (hunk 1 above)
Comment (siddancha) [reply python/sim.py:189]: I see, okay. As long as it doesn't add too much overhead.

File: python/sim.py (hunk 2 above)
Comment (siddancha) [reply python/sim.py:118]: Okay thanks!

**Review by siddancha — APPROVED** [review]
Looks good to me. Do you want to go ahead and merge?

**Pull request merged**